### PR TITLE
Adds working communications console that ends the round

### PR DESCRIPTION
**Pull request by Zumorica**
Just a very basic GUI with a countdown and a button that ends the round after X seconds.
I'm looking for some feedback. What else should I add to this? Is there anything I should change?
(Adding the actual emergency shuttle is out of the scope of this PR)

### CODE REVIEW

**Comment by PJB3005** [comment]
Definitely gonna need a way to *recall* it.

**Comment by Zumorica** [comment]
> Definitely gonna need a way to _recall_ it.

Done.

File: Content.Client/GameObjects/Components/Command/CommunicationsConsoleBoundUserInterface.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    public class CommunicationsConsoleBoundUserInterface : BoundUserInterface
+    {
+        [ViewVariables]
+        private CommunicationsConsoleMenu _menu;
+
+        public bool CountdownStarted { get; private set; }
+
+        public int Countdown => _expectedCountdownTime == null
+            ? 0 : Math.Max((int)(_expectedCountdownTime.Value.Subtract(DateTime.Now)).TotalSeconds, 0);
+        private DateTime? _expectedCountdownTime;
```
Comment (PJB3005) [inline Content.Client/GameObjects/Components/Command/CommunicationsConsoleBoundUserInterface.cs:19]: You should be using a `TimeSpan` for this, gotten from `IGameTiming.CurTime`.

File: Content.Server/GameObjects/EntitySystems/RoundEndSystem.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+{
+    public class RoundEndSystem : EntitySystem
+    {
+#pragma warning disable 649
+        [Dependency] private IGameTicker _gameTicker;
+#pragma warning restore 649
+
+        private CancellationTokenSource _roundEndCancellationTokenSource = new CancellationTokenSource();
+        public bool IsRoundEndCountdownStarted { get; private set; }
+        public int RoundEndCountdownTime { get; set; } = 5000;
```
Comment (PJB3005) [inline Content.Server/GameObjects/EntitySystems/RoundEndSystem.cs:18]: Store this as a `TimeSpan` or something.

Comment (PJB3005) [reply Content.Server/GameObjects/EntitySystems/RoundEndSystem.cs:18]: Also isn't 5 seconds kinda short?

File: Content.Client/GameObjects/Components/Command/CommunicationsConsoleBoundUserInterface.cs (hunk 1 above)
Comment (Zumorica) [reply Content.Client/GameObjects/Components/Command/CommunicationsConsoleBoundUserInterface.cs]: Done.

File: Content.Server/GameObjects/EntitySystems/RoundEndSystem.cs (hunk 2 above)
Comment (Zumorica) [reply Content.Server/GameObjects/EntitySystems/RoundEndSystem.cs]: Yeah, it was short on purpose for debugging purposes but I changed it to 4 minutes now. Also changed this to use TimeSpan.

File: Content.Server/GameObjects/EntitySystems/RoundEndSystem.cs [hunk 3]
```diff
@@ -34,7 +36,7 @@ public void RequestRoundEnd()
 
             IsRoundEndCountdownStarted = true;
 
-            ExpectedCountdownEnd = DateTime.Now.AddMilliseconds(RoundEndCountdownTime);
+            ExpectedCountdownEnd = _gameTiming.CurTime.Add(new TimeSpan(0,0,0,0,RoundEndCountdownTime));
```
Comment (PJB3005) [inline Content.Server/GameObjects/EntitySystems/RoundEndSystem.cs:39]:
```suggestion
            ExpectedCountdownEnd = _gameTiming.CurTime + TimeSpan.FromMilliseconds(RoundEndCountdownTime);
```

File: Content.Server/GameObjects/EntitySystems/RoundEndSystem.cs [hunk 4]
```diff
@@ -17,7 +17,7 @@ public class RoundEndSystem : EntitySystem
 
         private CancellationTokenSource _roundEndCancellationTokenSource = new CancellationTokenSource();
         public bool IsRoundEndCountdownStarted { get; private set; }
-        public int RoundEndCountdownTime { get; set; } = 5000;
+        public int RoundEndCountdownTime { get; set; } = 240000;
```
Comment (PJB3005) [inline Content.Server/GameObjects/EntitySystems/RoundEndSystem.cs:20]: See this is why I'd prefer this be stored as a `TimeSpan`, then you can just do
 ```cs
public TimeSpan RoundEndCountdownTime { get; set; } = TimeSpan.FromMinutes(4);
```

Comment (Zumorica) [reply Content.Server/GameObjects/EntitySystems/RoundEndSystem.cs]: Oh, that's much more convenient, alright! I changed it now.

**Pull request merged**